qprofiler2/src/org/qcmg/qprofiler2/summarise/KmersSummary.java

```diff
@@ -107,20 +97,7 @@ public static String producer(final int k, final String mers , final boolean inc
 		//produce all possible kmers in String 
 		String str1 = producer( k, "", includeN );		
 		return str1.split(Constants.COMMA_STRING);
```

Comment:
`TabTokenizer` is more efficient than `split`

Reply:
skipped

Reply:
I think it would be good to test to see how split performs compared to TabTokenizer. If you could produce some benchmarks, that would be great. Ideally using the string that a 6-mer would produce (should have around 35K substrings)

Reply:
the producer didn't call split and the method getPossibleKerString(..) only call it once. However this method is called 10 times in class BamSummaryReport, and 10 times in class FastqSumaryReport.  in the case of kmers6, there are lots of substrings after the split. 

I did experiment to call split and subString. It seems split is faster than subString with indexOf:
2019/01/31 13:41:00
calling producer with split 101 times 
number of kmers after split option: 4096
the finished producer with split 101 times 
2019/01/31 13:41:00
calling producer with subString 101 times 
number of kmers after substring option: 4096
2019/01/31 13:41:03
the finished producer with subString 100 times 

There are also some discussions online, after java7 split also calls indexOf. that's why people say split performance is close to subString. 
https://stackoverflow.com/questions/35388181/split-method-vs-substring-and-indexof